chance.js

```diff
@@ -479,6 +479,35 @@
         return new_array;
     };
 
+    // shuffle each passed in array symmetrically
+    // returns a collection of arrays
+    Chance.prototype.symmetricShuffle = function () {
+        var old_arrays = [];
+        var new_arrays = [];
+        for(var k = 0;k < arguments.length; k++) {
+            old_arrays.push(arguments[k].slice(0));
+            new_arrays.push([]);
+        }
+
+        var j = 0,
+            length = Number(old_arrays[0].length);
+
+        for (var i = 0; i < length; i++) {
+            // Pick a random index from the array
+            j = this.natural({max: old_arrays[0].length - 1});
+            for(var m = 0; m < old_arrays.length; m++ ) {
+                // Add it to the new arrays
+                new_arrays[m][i] = old_arrays[m][j];
+                // Remove that element from the original array
+                old_arrays[m].splice(j, 1);
```

Comment:
@somejeff, I don't know much about javascript runtime performance, but in other languages, like C++ or C#, the removal of elements from the middle of an array would have O(n) performance. Burying those operations inside a loop gives you O(n^2) performance for the entire algorithm.

This routine can be implemented with O(n) performance if you make a slight tweak

```
function range(size) {
    // reimplement lodash range
    return Array.apply(null, Array(size)).map(function (_, i) {return i;});
}

// inside your routine just in front of the loop
var sourceIndexes = shuffle(range(length));

        for (var i = 0; i < length; i++) {
            // Pick a random index from the array
            j = sourceIndexes[i];
            for(var m = 0; m < old_arrays.length; m++ ) {
                // Add it to the new arrays
                new_arrays[m][i] = old_arrays[m][j];
            }
        }
...
```

Reply:
Although in this case, the sourceIndexes is still calling the original shuffle function, which would also be O(n^2). 

I like your approach in creating an array by cherry picking entries from the original array.  Let me learn from that and see if I can tinker something up.

Reply:
I assumed that `shuffle` was already O(n), but after looking at the source, it has the same splice operation inside a loop. He could be redesigned to be O(n) using an exchange technique. I will see if I can put together a PR, and a benchmark that shows whether the change is really worth it.

Reply:
I put together a little benchmark. It is showing significant speed increase by eliminating the splice in shuffle. I will submit a PR for it.

```
var Chance = require('chance');
const chance = new Chance();

function range(size) {
    // reimplement lodash range
    return Array.apply(null, Array(size)).map(function (_, i) {return i;});
}

function shuffle(arr) {
    var new_array = [],
        j,
        length = Number(arr.length),
        source_indexes = range(length),
        last_source_index = length - 1,
        selected_source_index;

    for (var i = 0; i < length; i++) {
        // Pick a random index from the array
        selected_source_index = chance.natural({max: last_source_index});
        j = source_indexes[selected_source_index];

        // Add it to the new array
        new_array[i] = arr[j];

        // Mark the source index as used
        source_indexes[selected_source_index] = source_indexes[last_source_index];
        last_source_index -= 1;
    }

    return new_array;
};

const input = range(100000);

console.time('new shuffle');
for(var i=0; i<10; ++i);
    shuffle(input);
console.timeEnd('new shuffle');

console.time('old shuffle');
for(var i=0; i<10; ++i);
    chance.shuffle(input);
console.timeEnd('old shuffle');

Output:
new shuffle: 103.007ms
old shuffle: 2122.118ms
```

Reply:
@somejeff https://github.com/chancejs/chancejs/pull/353 to make `shuffle` O(n)